### ISSUE TITLE
iOS: Audio interrupted when handling datachannel events through the redux

### ISSUE THREAD

**Issue by pavigor** [react-native-webrtc/react-native-webrtc#1418]
I have a SFU server that sends some events through the datachannel. I have found that audio is interrupting when iOS client receives event. For example, server sends "talking" event and client should change participant's border from black to green and vice versa. Something like this:

```
const pc = new RTCPeerConnection({});
const dchannel = pc.createDataChannel("control");
dchannel.onmessage = (msg) => {
   dispatch(updateParticipantState(msg));
};
```

#### Expected Behavior:

Don't understand how datachannel affects the audio.
#### Observed Behavior:

Audio interrupted or can be delayed.

#### Platform Information  
* **React Native Version**:  0.71.2
* **WebRTC Module Version**:  111.0.3
* **Platform OS + Version**: iOS 15.7.6

**Comment by pavigor** [comment]
It would be great if somebody explain me how webrtc and audio render working under the hood. I know that React Native has three threads (UI, JS and Bridge). So I want to know in which thread WebRTC receives RTP and where is RTCView rendered ?

**Comment by saghul** [comment]
How many events per second are you sending?

The audio rendering happens automatically, without and RTCView and should not be affected by the JS thread being busy.

**Comment by saghul** [comment]
Is the snippet above all you are going? Or are you also calling `getStats` for measuring audio levels, for example?

**Comment by pavigor** [comment]
> Is the snippet above all you are going?

Yes, that's all. I'm not using `getStats`, instead of that I'm using extension from rfc6464 on the server side.

**Comment by saghul** [comment]
I honestly don't see how those 2 could be interacting since they run in different threads...

**Comment by pavigor** [comment]
> I honestly don't see how those 2 could be interacting since they run in different threads...

Could you please explain me some questions. When I debug my app I see webrtc logs in Metro console, this logs contains time in milliseconds (ms).

```
rn-webrtc:pc:DEBUG 19 ctor +1ms
rn-webrtc:pc:DEBUG 20 ctor +3ms
rn-webrtc:pc:DEBUG 20 ctor +397ms
```

What time field means in this logs? Sometimes I see over 500ms values.

**Comment by saghul** [comment]
Those timings refer to the previous log line. So the 20 creo happened 3ms after the previous log line.

**Comment by pavigor** [comment]
I have found the problem with my React component. There was `InCallManager.setSpeakerphoneOn(true);` on every re-render which caused the problem with audio.

Thank you for the help!
